docs/DeveloperGuide.md

```diff
@@ -1,34 +1,662 @@
-# Developer Guide
+# CCA Manager Developer Guide
 
-## Design & implementation
+![](CCAManager_logo.png)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+# Table of Contents
 
+- [1. Introduction](#1-introduction)
+- [2. Setting up](#2-setting-up)
+- [3. Design and Implementation](#3-design-and-implementation)
+  * [3.1. Input Parsing](#31-input-parsing)
+  * [3.2. Commands](#32-commands)
+  * [3.3. Finance](#33-finance)
+  * [3.4. Event](#34-event)
+  * [3.5. HR](#35-hr)
+  * [3.6. Storage](#36-storage)
+- [4. Product scope](#4-product-scope)
+  * [4.1. Target user profile](#41-target-user-profile)
+  * [4.2. Value proposition](#42-value-proposition)
+- [5. User Stories](#5-user-stories)
+- [6. Non-Functional Requirements](#6-non-functional-requirements)
+- [7. Glossary](#7-glossary)
+- [8. Instructions for manual testing](#8-instructions-for-manual-testing)
 
-## Product scope
-### Target user profile
 
-{Describe the target user profile}
+## 1. Introduction 
+CCA Manager is a revolutionary all-in-one management tool that changes the way you can manage interest groups with unrivaled efficiency and simplicity. Its lightweight Command Line Interface (CLI) allows administrators to breeze through tasks quickly and easily while offering powerful features to advanced users.
 
-### Value proposition
+This developer guide is written to document the implementation of CCA Manager. This document is intended for people who
+are interested in learning more about the technical details of the various features and the organization of the application.
 
-{Describe the value proposition: what problem does it solve?}
+## 2. Setting up
+Refer to the guide here.
 
-## User Stories
+## 3. Design and Implementation
+This section seeks to explain the high-level design of the application. Given below is a quick overview of each component and the explanation of the design architecture in greater detail. 
+Diagrams found in our documentation were generated using PlantUML and references were made to addressbook-level2 for the structure of the classes and packages. The structures have been modified to meet the needs of our application.
+
+### 3.1. Input Parsing
+![Parser](BackendDiagram/ParserFlow.png)
+
+Input parsing describes the process of converting the user's input into an executable command. The diagram above shows the execution flow required to run a single command.
```

Comment:
Maybe the diagram could be put after describing the figure in here? As it may be confusing for readers to see the diagram first and then see the explanation